cmd/pipeline/deploy.go

```diff
@@ -249,6 +253,27 @@ func (pc *Command) ExecuteDeployPipeline(ctx context.Context, opts *DeployOption
 	return nil
 }
 
+// shouldRedeployDependencies determines if the pipeline should redeploy dependencies
+// default behavior is set by cluster config, but can be overridden by the user using the flag --dependencies
+func shouldRedeployDependencies(opts *DeployOptions) bool {
+	isForceRedeployDependenciesSetInOktetoInstance := env.LoadBoolean(constants.OktetoForceRedeployDependencies)
+	isInsideDependency := env.LoadBoolean(constants.OktetoIsDependencyEnvVar)
+
+	if isInsideDependency {
+		return false
+	}
+
+	if !isInsideDependency && isForceRedeployDependenciesSetInOktetoInstance {
```

Comment:
```suggestion
	if isForceRedeployDependenciesSetInOktetoInstance {
```

This is not needed as here, we can say it is not inside a dependency because if not, it would have returned in line 263

In fact, I think this can be simplified to avoid different indentation levels:

```golang
func shouldRedeployDependencies(opts *DeployOptions) bool {
	isForceRedeployDependenciesSetInOktetoInstance := env.LoadBoolean(constants.OktetoForceRedeployDependencies)
	isInsideDependency := env.LoadBoolean(constants.OktetoIsDependencyEnvVar)

	if isInsideDependency {
		return false
	}

	if !isForceRedeployDependenciesSetInOktetoInstance {
		return opts.RedeployDependencies
	}

	// the user forces --dependencies=false
	if opts.DependenciesIsSet && !opts.RedeployDependencies {
		return false
	}
	return true
}
```